### ISSUE TITLE
Error "Use of 'mgl_interpolate:withCurveType:parameters:stops:' as an NSExpression is forbidden" on iOS 15

### ISSUE THREAD

**Issue by adetante** [maplibre/maplibre-navigation-ios#6]
### Steps to Trigger Behavior

1. Use the provided [sample code](https://github.com/maplibre/maplibre-navigation-ios#sample-code)
2. Call to `navigationView.showRoutes` causes the error

### Actual Behavior

- Error:
```
NSPredicate: Use of 'mgl_interpolate:withCurveType:parameters:stops:' as an NSExpression function is forbidden
```
- Map is not displayed

### Version(s) affected

- Platform: iOS
- OS version: iOS 16
- Device type: iPhone 14 Pro
- Emulator/ Simulator: yes/yes
- Development OS: OSX 13.5.2
- maplibre-navigation-ios Version 2.0.0
- MapLibre GL version 5.13.0

### Additional context

- This is fixed into `maplibre-native` library: https://github.com/maplibre/maplibre-native/pull/411
- Here is the related issue on `maplibre-native`: https://github.com/maplibre/maplibre-native/issues/331
- Here is the documentation of the required migration: https://maplibre.org/maplibre-native/ios/api/migrating-to-expressions.html#interpolation-mode:
```
if #available(iOS 15, *) {
    circleExpression = NSExpression(
        forMGLInterpolating: NSExpression(forConstantValue: mag),
        curveType: .linear,
        parameters: nil,
        stops: NSExpression(forConstantValue: stops))
} else {
    // This works up to iOS 14.5
    circleExpression = NSExpression(
        format: "mgl_interpolate:withCurveType:parameters:stops:(mag, 'linear', nil, %@)",
        stops)
}
```

**Comment by Djibs** [comment]
I confirm

**Comment by Patrick-Kladek** [comment]
I think we can close this as it was resolved in #33

**Comment by boldtrn** [comment]
Ok, please reopen if this occurs again 👍
